components/console-backend-service/internal/domain/rafter/asset_service_test.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	AssetNamespace = "AssetNamespace"
+)
+
+func TestAssetService_Find(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		asset1 := fixSimpleAsset("1", nil)
+		asset2 := fixSimpleAsset("2", nil)
+		asset3 := fixSimpleAsset("3", nil)
+
+		service := fixFakeAssetService(t, asset1, asset2, asset3)
+
+		result, err := service.Find(AssetNamespace, "1")
+		require.NoError(t, err)
+		assert.Equal(t, asset1, result)
+	})
+
+	t.Run("NotFound", func(t *testing.T) {
+		service := fixFakeAssetService(t)
+
+		result, err := service.Find(AssetNamespace, "1")
+		require.NoError(t, err)
+		assert.Nil(t, result)
+	})
+}
+
+func TestAssetService_ListForAssetGroupByType(t *testing.T) {
+	t.Run("Success with assetGroupName paramater", func(t *testing.T) {
+		asset1 := fixSimpleAsset("1", map[string]string{
+			rafter.AssetGroupLabel: "exampleAssetGroupA",
+		})
+		asset2 := fixSimpleAsset("2", map[string]string{
+			rafter.AssetGroupLabel: "exampleAssetGroupB",
+		})
+		asset3 := fixSimpleAsset("3", map[string]string{
+			rafter.AssetGroupLabel: "exampleAssetGroupC",
+		})
+		expected := []*v1beta1.Asset{asset1}
+
+		service := fixFakeAssetService(t, asset1, asset2, asset3)
+
+		result, err := service.ListForAssetGroupByType(AssetNamespace, "exampleAssetGroupA", nil)
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+	})
+
+	t.Run("Success with whole paramaters", func(t *testing.T) {
+		asset1 := fixSimpleAsset("1", map[string]string{
+			rafter.AssetGroupLabel: "exampleAssetGroupA",
+			rafter.TypeLabel:       "markdown",
+		})
+		asset2 := fixSimpleAsset("2", map[string]string{
+			rafter.AssetGroupLabel: "exampleAssetGroupB",
+			rafter.TypeLabel:       "json",
+		})
+		asset3 := fixSimpleAsset("3", map[string]string{
+			rafter.AssetGroupLabel: "exampleAssetGroupC",
+			rafter.TypeLabel:       "yaml",
+		})
+		expected := []*v1beta1.Asset{asset1}
+
+		service := fixFakeAssetService(t, asset1, asset2, asset3)
+
+		result, err := service.ListForAssetGroupByType(AssetNamespace, "exampleAssetGroupA", []string{"markdown"})
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+	})
+
+	t.Run("NotFound", func(t *testing.T) {
+		service := fixFakeAssetService(t)
+
+		result, err := service.ListForAssetGroupByType(AssetNamespace, "exampleDocsTopic", nil)
+		require.NoError(t, err)
+		assert.Nil(t, result)
+	})
+}
+
+func TestAssetService_Subscribe(t *testing.T) {
+	t.Run("Simple", func(t *testing.T) {
+		service := fixFakeAssetService(t)
+
+		assetListener := listener.NewAsset(nil, nil, nil)
+		service.Subscribe(assetListener)
+	})
+
+	t.Run("Duplicated", func(t *testing.T) {
+		service := fixFakeAssetService(t)
+
+		assetListener := listener.NewAsset(nil, nil, nil)
+		service.Subscribe(assetListener)
+		service.Subscribe(assetListener)
```

Comment:
line repeated

Reply:
Yes exactly, this test tests duplicated subscriptions.